yt_dlp/extractor/itprotv.py

```diff
@@ -0,0 +1,173 @@
+                course_name, headers=headers, note='Fetching data from course API')
+
+        return course_api
+
+    def _get_episode_api_json(self, webpage, episode_id):
+        jwt = self._fetch_jwt(webpage)
+
+        if jwt:
+            headers = {'Authorization': 'Bearer ' + jwt}
+            episode_api = self._download_json(
+                f'https://api.itpro.tv/api/urza/v3/consumer-web/brand/00002560-0000-3fa9-0000-1d61000035f3/episode?url={episode_id}',
+                episode_id, headers=headers, note='Fetching data from episode API')
+
+        return episode_api
+
+    def _fetch_jwt(self, webpage):
+        return self._search_regex(r'{"passedToken":"([A-Za-z0-9-_]*\.[A-Za-z0-9-_]*\.[A-Za-z0-9-_]+)",', webpage, 'jwt')
+
+    def _check_if_logged_in(self, webpage):
+        if re.match(r'{\s*member\s*:\s*null', webpage):
+            raise self.raise_login_required()
+
+    def _real_extract(self, url):
+        QUALITIES = qualities(['low', 'medium', 'high', 'veryhigh'])
+
+        episode_id = self._match_id(url)
+        webpage = self._download_webpage(url, episode_id)
+        self._check_if_logged_in(webpage)
+
+        course_name = re.match(r'https?://.*/course/(?P<course_name>[0-9a-z-]+)/.*', url).group(1)
+        course = self._get_course_api_json(webpage, course_name)['course']
+
+        episode = self._get_episode_api_json(webpage, episode_id)['episode']
+
+        for i in range(len(course.get('topics'))):
+            if traverse_obj(course, ('topics', i, 'id'), expected_type=str) == episode.get('topic'):
+                chapter_name = traverse_obj(course, ('topics', i, 'title'), expected_type=str)
+                chapter_id = traverse_obj(course, ('topics', i, 'id'), expected_type=str)
+                chapter_number = i + 1
+
+        return {
+            'id': episode_id,
+            'title': episode.get('title'),
+            'description': episode.get('description'),
+            'thumbnail': episode.get('thumbnail'),
+            'formats': [
+                {'url': episode[f'jwVideo{h}Embed'], 'height': h}
+                for h in (320, 480, 720, 1080) if episode.get(f'jwVideo{h}Embed')
+            ],
+            'duration': int_or_none(episode.get('length')),
+            'series': course.get('name'),
+            'series_id': course.get('url'),
+            'availability': self._availability(
+                needs_premium=False, needs_subscription=False, needs_auth=True,
+                is_private=False, is_unlisted=False),
+            'chapter': chapter_name,
+            'chapter_number': chapter_number,
+            'chapter_id': chapter_id,
+            'subtitles': {
+                'en': [{'ext': 'vtt', 'data': episode['enCaptionData']}]
+            } if episode.get('enCaptionData') else None,
+        }
+
+
+class ITProTVCourseIE(ITProTVIE):
+    _VALID_URL = r'https?://app.itpro.tv/course/(?P<id>[0-9a-z-]+)'
```

Comment:
```suggestion
    _VALID_URL = r'https?://app.itpro.tv/course/(?P<id>[0-9a-z-]+)/?(?:$|[#?])'
```
should fix the `test_all_urls` issue